[llm] Update vllm to 0.11.0 and Nixl to 0.6.0

kouroshHakha opened the pull request:
Updating vLLM and Nixl -- notable changes: 
- Remove V0 from stack
- Embedding is added to V1. So it should be ok

kouroshHakha committed:
wip

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>

kouroshHakha committed:
wip

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>

kouroshHakha committed:
wip

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>

kouroshHakha committed:
wip

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>

kouroshHakha committed:
log_engine_metrics by default

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>

nrghosh committed:
wip - fix test import

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

nrghosh reviewed (commented):
- [x]  unit tests fixed

- [x] release tests (probes/test_basic.py::test_logprobs)
logprobs problem: -1 is being interpreted as a vocab size instead of an invalid # for logprobs so all release tests are barfing b/c that probe test fails. 

- [x] release tests 1p1d / 2p6d faliing due to nixl error 

`TypeError: nixl_agent_config.__init__() got an unexpected keyword argument 'num_threads'`

`num_threads` was added to `nixl_agent_config` between [release/0.4.1](https://github.com/ai-dynamo/nixl/blob/release/0.4.1/src/api/python/_api.py#L45-L56) and [release/0.6.0](https://github.com/ai-dynamo/nixl/blob/release/0.6.0/src/api/python/_api.py#L134-L148) at [9f77cc4](https://github.com/ai-dynamo/nixl/commit/9f77cc47dbd03b8421f9cfd393273388e6e7c3bb)

**Issue**: release env is still using `nixl=0.4.1` which is causing the break
<img width="749" height="317" alt="image" src="https://github.com/user-attachments/assets/645c9aaa-0141-4b28-857f-14eb964d2fbc" />


**New issue** - nixl linker / backend error

`failed to load plugin from /usr/local/nixl/lib/x86_64-linux-gnu/plugins/libplugin_UCX_MO.so: /usr/local/nixl/lib/x86_64-linux-gnu/plugins/libplugin_UCX_MO.so: undefined symbol: _ZN12nixlDescListI12nixlMetaDescEC1ERK10nixl_mem_tRKbRKi`

```
(RayWorkerWrapper pid=128409) INFO 10-07 11:58:24 [factory.py:51] Creating v1 connector with name: NixlConnector and engine_id: 16561289-139e-44af-90e7-1c86a61925b2-10.0.111.115-40773
(RayWorkerWrapper pid=128409) INFO 10-07 11:58:24 [nixl_connector.py:465] Initializing NIXL wrapper
(RayWorkerWrapper pid=128409) INFO 10-07 11:58:24 [nixl_connector.py:466] Initializing NIXL worker 16561289-139e-44af-90e7-1c86a61925b2-10.0.111.115-40773
(RayWorkerWrapper pid=128409) E1007 11:58:24.959858  128409 nixl_plugin_manager.cpp:122] Failed to load plugin from /usr/local/nixl/lib/x86_64-linux-gnu/plugins/libplugin_UCX_MO.so: /usr/local/nixl/lib/x86_64-linux-gnu/plugins/libplugin_UCX_MO.so: undefined symbol: _ZN12nixlDescListI12nixlMetaDescEC1ERK10nixl_mem_tRKbRKi
(RayWorkerWrapper pid=128409) E1007 11:58:24.959885  128409 nixl_plugin_manager.cpp:288] Failed to load plugin 'UCX_MO' from any directory
```

**Resolved by reinstalling nixl (local issue)**

nrghosh committed:
fix probes (release) test test_basic.py::test_logprobs

- https://github.com/vllm-project/vllm/pull/23868
- PR in vLLM changed interpretation of num_logprobs = -1
- Overrides to model_config.get_vocab_size(), which triggers
  openai.APIError instead of openai.badRequestError
- Test expects the latter, which causes failure
- Instead of broadening / changing the expected failure type, we use -2

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

nrghosh committed:
update nixl in llm-requirements and re-compile deps

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>
Signed-off-by: nikhil <nikhil@anyscale.com>

nrghosh committed:
Merge branch 'master' into kh/update-vllm-0.11.0

aslonnie reviewed (commented):
can we wait till 2.50 branch cut to merge this?

nrghosh commented on python/ray/llm/_internal/batch/stages/vllm_engine_stage.py:
```diff
@@ -330,53 +323,7 @@ async def generate_async(
-        )
-
-    async def generate_async_v1(self, request: vLLMEngineRequest) -> Any:
+    async def _generate_async(self, request: vLLMEngineRequest) -> Any:
```
it's a method, not an instance attribute - doesn't need init
called from line 319 `(self._generate_async(request)) `

nrghosh commented:
> can we wait till 2.50 branch cut to merge this?
I think we want to get this in ASAP, it has a lot of performance improvements 

cc @kouroshHakha

eicherseiji commented:
> can we wait till 2.50 branch cut to merge this?
If we ship the switch to OpenTelemetry without vLLM 0.11.0 we will have no vLLM metrics in Ray 2.50. cc: @can-anyscale

nrghosh reviewed (approved):
Unit tests and release tests both passing 

Doc build failure seems ephemeral / unrelated, as full doc build succeeds locally with this branch.
